report_py3o_fusion_server/models/ir_actions_report.py

```diff
@@ -0,0 +1,44 @@
+# © 2013 XCG Consulting <http://odoo.consulting>
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+from openerp import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+logger = logging.getLogger(__name__)
+
+try:
+    from py3o.formats import Formats
+except ImportError:
+    logger.debug('Cannot import py3o.formats')
+
+
+class IrActionsReport(models.Model):
+    _inherit = 'ir.actions.report'
+
+    @api.multi
+    @api.constrains("py3o_is_local_fusion", "py3o_server_id", "py3o_filetype")
+    def _check_py3o_server_id(self):
+        for report in self:
+            if report.report_type != "py3o":
+                continue
+            is_native = Formats().get_format(report.py3o_filetype).native
+            if ((not is_native or not report.py3o_is_local_fusion) and
```

Comment:
```suggestion
            if (not is_native 
                    and not report.py3o_is_local_fusion and
```

We need to change this conditions, it is impossible to create new reports with local fusion when `report_py3o_fusion_server` is installed. Put another way the previous condition means all new reports created must have  a fusion server defined when `report_py3o_fusion_server` which is not practical.

Reply:
@lmignon @hbrunn may I summon your opinion on this one? Thanks in advance.

Reply:
in 10 it looks like this: https://github.com/OCA/reporting-engine/blob/10.0/report_py3o_fusion_server/models/ir_actions_report_xml.py#L27 - so at least it's consistent ;-)

Can't we just leave out the is_native check entirely? If it's ie a `.doc` and there's no rendering server, the libreoffice code path will take care of it I think, right? How about some tests for the different cases?

Reply:
ah, now I understand my own code again. Check out https://github.com/OCA/reporting-engine/blob/b81219abd33d56e6068212d5bb10587b06d3db95/report_py3o_fusion_server/models/py3o_report.py#L35 - this is where those things are relevant, and the idea is that you tick the local fusion box when you have the server module installed but you want to force libreoffice conversion instead of the server. So I rectify and now this the code is fine as it is

Reply:
@hbrunn I recall now that local fusion and local format conversion is two different things. You can do local fusion, and use the py3o server to do the format conversion. Or delegate fusion to the py3o server too.

So yeah, the code is probably fine as is.

Yet we still have a problem. Let me explain our situation.
- we have an Odoo database with report_py3o_fusion_server installed
- we try to install a new module that attempts to create a new py3o PDF report; that module does not know about report_py3o_fusion_server, because it is normally used with local conversion, so it does not attach the report to a fusion server => that installation fails because the report is not native and there is no server

Not sure what to do... may be have a notion of default fusion server that is automatically set when creating new non native report when report_py3o_fusion_server is installed?

Reply:
hmmm. Make it an onchange warning maybe? Or default local_fusion to true?

Reply:
@hbrunn @sbidoul The default for `py3o_is_local_fusion` is `True`. The problem is that before the split, report_py3o/report_py3O_fusion_server,  we assumed that there was no libreoffice runtime available on the machine. Therefore a py3o_server was required to transform the result of the fusion into the expected format if not native. IMO, we should have a way to detect into py3o_report that the libreoffice runtime is available and raise this error only if the runtime is not available.

Reply:
@sbidoul and @hbrunn I've implemented a non blocking approach of this problem. In short:
* The constrains only fails if not local fusion and not fursion server
* We log at server startup an display a warning message on the report if the report should not work due to the unavailability of the libreoffice runtime or a fusion server
* We raise a proper error message when the libreoffice runtime is called to generate a report and is not available.

With these changes the installation of new py3o reports in non native format should work even if py3o_report_fusion_server is installed and a libreoffice runtime is not available...